### PR TITLE
Compile for generic x86-64 which is compatible with M1 and M2

### PR DESCRIPTION
**Pull request by jelmervdl**
Basically taken from https://stackoverflow.com/questions/69803659/what-is-the-proper-way-to-build-for-macos-x86-64-using-cmake-on-apple-m1-arm but I tested a local build with these options and it worked on an M2 Macbook Air.

### CODE REVIEW

**Comment by XapaJIaMnu** [comment]
And it translated? We don't have generic intgemm path for it?

**Comment by XapaJIaMnu** [comment]
Should we add a generic x86 build for windows and ubuntu and label them compatibility?

**Comment by jelmervdl** [comment]
I'm not entirely sure which Intgemm path it takes… but yes it works properly:
189628664-5a90b585-2919-4927-8622-191bc04f8b51

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -117,7 +120,7 @@ jobs:
       # and build directories, but this is only available with CMake 3.13 and higher.  
       # The CMake binaries on the Github Actions machines are (as of this writing) 3.12
       run: |-
-        cmake $GITHUB_WORKSPACE -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DBUILD_ARCH=${{ matrix.arch }} ${{ env.ccache_cmake }}
+        cmake $GITHUB_WORKSPACE -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DBUILD_ARCH=${{ matrix.arch }} -DCMAKE_OSX_ARCHITECTURES=x86_64 ${{ env.ccache_cmake }}
```
Comment (XapaJIaMnu) [inline .github/workflows/build.yml:123]: Is this necessary on top of build arch?

**Pull request merged**